### PR TITLE
docs: Add and fix links in free-threading guide.

### PR DESCRIPTION
**Pull request by ngoldbaum**
This is mostly adding links. There are a few small text edits and fixes as well.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
I could probably use `PYO3_DOCS_URL` instead of `latest` in all the links to the API docs. It looks like there's only one existing link in the guide that does that, maybe I should mass-convert all the links to the API docs to do that, except if they need to point to a specific version like in the migration guide.

**Comment by davidhewitt** [comment]
Yes please, I think it would be better to use the links to the exact PyO3 version using `PYO3_DOCS_URL`. 👍 

Other than that, this is a welcome improvement, thanks 🙏

**Comment by ngoldbaum** [comment]
Moved all the `/latest/` links to use the `PYO3_DOCS_URL` pattern. Also rebased and added links to the docs I wrote for `OnceExt` and `OnceLockExt` in another PR.

File: guide/src/migration.md [hunk 1]
```diff
@@ -1064,7 +1064,7 @@ Python::with_gil(|py| {
...
+Furthermore, `Python::acquire_gil` provides ownership of a `GILGuard` which can be freely stored and passed around. This is usually not helpful as it may keep the lock held for a long time thereby blocking progress in other parts of the program. Due to the generative lifetime attached to the GIL token supplied by `Python::with_gil`, the problem is avoided as the GIL token can only be passed down the call chain. Often, this issue can also be avoided entirely as any GIL-bound reference `&'py PyAny` implies access to a GIL token `Python<'py>` via the [`PyAny::py`](https://docs.rs/pyo3/0.22.5/pyo3/types/struct.PyAny.html#method.py) method.
```
Comment (ngoldbaum) [inline guide/src/migration.md:1067]: this one is a method that will be removed in 0.23

File: guide/pyclass-parameters.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ struct MyClass {}
 struct MyClass {}
 ```
 
-[params-1]: https://docs.rs/pyo3/latest/pyo3/types/struct.PyAny.html
+[params-1]: {{#PYO3_DOCS_URL}}/pyo3/types/struct.PyAny.html
```
Comment (davidhewitt) [inline guide/pyclass-parameters.md:42]: This file is actually reused inside `lib.rs` so can't have this placeholder :(

```suggestion
[params-1]: https://docs.rs/pyo3/latest/pyo3/types/struct.PyAny.html
```

**Review by davidhewitt — APPROVED** [review]
Super, please merge after reverting just one piece :)

**Pull request merged**